libminifi/test/TestBase.h

```diff
@@ -164,6 +165,31 @@ class LogTestController {
     return found;
   }
 
+  std::optional<std::smatch> matchesRegex(const std::string &regex_str,
```

Comment:
as this is a single use function, it would be better IMO if we moved it into the `AttributesToJSONTests.cpp` if possible, thus other test wouldn't have to parse `<regex>` (or move the impl to the `TestBase.cpp` and return a `vector<string>`, but that would sacrifice some semantic power in the name of build-time)

Reply:
The `matchesRegex` depends on the `log_output` member of the `LogTestController` to retrieve and match the log output, and there is no way currently to get the `log_output` outside of the class. We could check how it influences the build times of other tests, but if it's not that big impact, I would keep it as a utility for other use cases as well, and would not expose the `log_output`.

Reply:
measured an unrelated test to see if this method makes any difference, fortunately the difference is negligible (~50ms) so I think we can keep it as is

Reply:
Thanks for the effort :)